ibm/service/power/data_source_ibm_pi_network_address_groups.go

```diff
@@ -0,0 +1,138 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+	"log"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func DataSourceIBMPINetworkAddressGroups() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPINetworkAddressGroupsRead,
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			// Attributes
+			Attr_NetworkAddressGroups: {
+				Computed:    true,
+				Description: "list of Network Address Groups.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_CRN: {
+							Computed:    true,
+							Description: "The Network Address Group's crn.",
+							Type:        schema.TypeString,
+						},
+						Attr_ID: {
+							Computed:    true,
+							Description: "The id of the Network Address Group.",
+							Type:        schema.TypeString,
+						},
+						Attr_Members: {
+							Computed:    true,
+							Description: "The list of IP addresses in CIDR notation (for example 192.168.66.2/32) in the Network Address Group.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									Attr_CIDR: {
+										Computed:    true,
+										Description: "The IP addresses in CIDR notation for example 192.168.1.5/32.",
+										Type:        schema.TypeString,
+									},
+									Attr_ID: {
+										Computed:    true,
+										Description: "The id of the Network Address Group member IP addresses.",
+										Type:        schema.TypeString,
+									},
+								},
+							},
+							Type: schema.TypeList,
+						},
+						Attr_Name: {
+							Computed:    true,
+							Description: "The name of the Network Address Group.",
+							Type:        schema.TypeString,
+						},
+					},
```

Comment:
I think the Attr_UserTags itself is missing still.

Reply:
Not needed here